core/primitives/src/bandwidth_scheduler.rs

```diff
@@ -106,3 +110,138 @@ pub struct BandwidthSchedulerState {
     /// Random data for now
     pub mock_data: [u8; 32],
 }
+
+/// Parameters used in the bandwidth scheduler algorithm.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub struct BandwidthSchedulerParams {
+    /// This much bandwidth is granted by default.
+    pub base_bandwidth: Bandwidth,
+    /// The maximum amount of data that a shard can send or receive at a single height.
+    pub max_shard_bandwidth: Bandwidth,
+    /// Maximum size of a single receipt.
+    pub max_receipt_size: Bandwidth,
+    /// Maximum bandwidth allowance that a link can accumulate.
+    pub max_allowance: Bandwidth,
+}
+
+impl BandwidthSchedulerParams {
+    pub fn calculate_from_config(
+        num_shards: NonZeroU64,
+        runtime_config: &RuntimeConfig,
+    ) -> BandwidthSchedulerParams {
+        // TODO(bandwidth_scheduler) - put these parameters in RuntimeConfig.
+        let max_shard_bandwidth: Bandwidth = 4_500_000;
+        let max_allowance = max_shard_bandwidth;
+        let max_base_bandwidth = 100_000;
+
+        let max_receipt_size = runtime_config.wasm_config.limit_config.max_receipt_size;
+
+        if max_shard_bandwidth < max_receipt_size {
+            panic!(
+                "max_shard_bandwidth is smaller than max_receipt_size! ({} < {}).
+                Invalid configuration - it'll be impossible to send a maximum size receipt.",
+                max_shard_bandwidth, max_receipt_size
+            );
+        }
+
+        // A receipt with maximum size must still be able to get through
+        // after base bandwidth is granted to everyone. We have to ensure that:
+        // base_bandwidth * num_shards + max_receipt_size <= max_shard_bandwidth
+        let available_bandwidth = max_shard_bandwidth - max_receipt_size;
+        let mut base_bandwidth = available_bandwidth / num_shards;
+        if base_bandwidth > max_base_bandwidth {
+            base_bandwidth = max_base_bandwidth;
+        }
```

Comment:
mini nit: 
let base_bandwidth = base_bandwidth.min(max_base_bandwidth)

Reply:
I like the explicit logic in the current version, IMO the one liner is harder to understand, min, max, requires more cognitive load.